### ISSUE TITLE
Documentation for `build_association`

### ISSUE THREAD

**Issue by ioquatix** [rails/rails#33599]
What does https://guides.rubyonrails.org/association_basics.html#methods-added-by-belongs-to-build-association-attributes do if an associated value already exists?

Does it delete it? Or does it modify the existing one?

Can we clarify the documentation?

**Comment by ioquatix** [comment]
I think this documentation could be improved for `has_one` too.

**Comment by prithvi16** [comment]
I am a jr developer. Can I take this up if it is approved?

**Comment by ioquatix** [comment]
Feel free to have a go at it!

**Comment by tgturner** [comment]
I wouldn't think more documentation would be needed for `build_association` the documentation already states, "This object will be instantiated from the passed attributes, and the link through this object's foreign key will be set, but the associated object will not yet be saved."

This method does not do anything at the DB level, and doesn't need to say anything about what happens when you call `save` on the built association.

**Comment by rafaelfranca** [comment]
Agree. When it is saved it will just update the foreign key and create a new record. No modification or deletion of records happen. Also this is already explained in the belongs_to documentation. http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-belongs_to

The deletion happens on the has_one case and it is also explained in the has_one documentation.

**Comment by ioquatix** [comment]
Adding "No modification or deletion of records happen." to `has_one` `association=` would be perfect.

**Comment by ioquatix** [comment]
Haha, sorry, I was so confused. Please forgive me. 

I meant to say:

> Adding "No modification or deletion of existing records happen." to `belongs_to` `association=` would be perfect.

**Comment by rafaelfranca** [comment]
Ahh, yeah, that one we should do. I'll reopen. @prithvi16 wants to work on adding that information to the doc?

**Comment by prithvi16** [comment]
Yes. Will start right away. Thanks 😃

**Comment by bogdanvlviv** [comment]
This issue can be closed since #33857 was merged.
